csxtools/utils.py

```diff
@@ -76,20 +77,28 @@ def get_fastccd_images(light_header, dark_headers=None,
 
         bgnd = np.array(dark)
 
-    data = _get_images(light_header, tag)
-    data = correct_images(data, bgnd, gain=gain)
+    data, n_images = _get_images(light_header, tag)
+    data = correct_images(data, bgnd, flat=flat, gain=gain)
     data = rotate90(data, 'cw')
-    return data
+    return data, n_images
```

Comment:
@stuwilkins @danielballan @danielballan 
See above note. Another alternative (which Tom seems to favor from our brief discussion), is to process images by events instead of all at once. Either way it is done, I think it is useful to be able to group images by events.